django_datawatch/base.py

```diff
@@ -131,17 +131,26 @@ def get_form_class(self):
 
     def save(self, payload, status, data=None, unacknowledge=False):
         # build default data
-        defaults = dict(status=status, data=data, assigned_to_user=self.get_assigned_user(payload, status),
-                        assigned_to_group=self.get_assigned_group(payload, status),
-                        payload_description=self.get_payload_description(payload))
+        defaults = dict(status=status, data=data, payload_description=self.get_payload_description(payload))
+
         if unacknowledge:
             defaults.update(dict(acknowledged_by=None, acknowledged_at=None, acknowledged_until=None))
 
-        with track_status_history(self.slug, self.get_identifier(payload), status):
-            # save the check
-            dataset, created = Result.objects.update_or_create(
-                slug=self.slug, identifier=self.get_identifier(payload),
-                defaults=defaults)
+        with transaction.atomic():
+            with track_status_history(self.slug, self.get_identifier(payload), status):
+                # save the check
+                dataset, created = Result.objects.update_or_create(
+                    slug=self.slug, identifier=self.get_identifier(payload),
+                    defaults=defaults)
+
+                # set assigned users and groups
+                if groups := self.get_assigned_groups(payload, status):
+                    assert len(groups) == len(set(g.pk for g in groups)), 'groups must be unique'
```

Comment:
I'd also be fine with letting the exception bubble through as customizing a check to yield the same group twice would be considered a misuse of the application logic.